src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -0,0 +1,675 @@
+  ctx->store_out_mcache = store_out->mcache;
+  ctx->store_out_sync   = fd_mcache_seq_laddr( ctx->store_out_mcache );
+  ctx->store_out_depth  = fd_mcache_depth( ctx->store_out_mcache );
+  ctx->store_out_seq    = fd_mcache_seq_query( ctx->store_out_sync );
+  ctx->store_out_chunk0 = fd_dcache_compact_chunk0( fd_wksp_containing( store_out->dcache ), store_out->dcache );
+  ctx->store_out_mem    = topo->workspaces[ topo->objs[ store_out->dcache_obj_id ].wksp_id ].wksp;
+  ctx->store_out_wmark  = fd_dcache_compact_wmark( ctx->store_out_mem, store_out->dcache, store_out->mtu );
+  ctx->store_out_chunk  = ctx->store_out_chunk0;
+
+  fd_memcpy( ctx->src_mac_addr, tile->repair.src_mac_addr, 6 );
+
+  /* Set up contact info tile input */
+  fd_topo_link_t * contact_in_link   = &topo->links[ tile->in_link_id[ CONTACT_IN_IDX ] ];
+  ctx->contact_in_mem    = topo->workspaces[ topo->objs[ contact_in_link->dcache_obj_id ].wksp_id ].wksp;
+  ctx->contact_in_chunk0 = fd_dcache_compact_chunk0( ctx->contact_in_mem, contact_in_link->dcache );
+  ctx->contact_in_wmark  = fd_dcache_compact_wmark ( ctx->contact_in_mem, contact_in_link->dcache, contact_in_link->mtu );
+
+  /* Set up tile stake weight tile input */
+  fd_topo_link_t * stake_weights_in_link   = &topo->links[ tile->in_link_id[ STAKE_IN_IDX ] ];
+  ctx->stake_weights_in_mem    = topo->workspaces[ topo->objs[ stake_weights_in_link->dcache_obj_id ].wksp_id ].wksp;
+  ctx->stake_weights_in_chunk0 = fd_dcache_compact_chunk0( ctx->stake_weights_in_mem, stake_weights_in_link->dcache );
+  ctx->stake_weights_in_wmark  = fd_dcache_compact_wmark ( ctx->stake_weights_in_mem, stake_weights_in_link->dcache, stake_weights_in_link->mtu );
+
+  /* Set up tile repair request input */
+  fd_topo_link_t * repair_req_in_link = &topo->links[ tile->in_link_id[ STORE_IN_IDX ] ];
+  ctx->repair_req_in_mem    = topo->workspaces[ topo->objs[ repair_req_in_link->dcache_obj_id ].wksp_id ].wksp;
+  ctx->repair_req_in_chunk0 = fd_dcache_compact_chunk0( ctx->repair_req_in_mem, repair_req_in_link->dcache );
+  ctx->repair_req_in_wmark  = fd_dcache_compact_wmark ( ctx->repair_req_in_mem, repair_req_in_link->dcache, repair_req_in_link->mtu );
+
+  /* Valloc setup */
+  void * alloc_shalloc = fd_alloc_new( alloc_shmem, 3UL );
+  if( FD_UNLIKELY( !alloc_shalloc ) ) { 
+    FD_LOG_ERR( ( "fd_allow_new failed" ) ); }
+  fd_alloc_t * alloc = fd_alloc_join( alloc_shalloc, 3UL );
+  if( FD_UNLIKELY( !alloc ) ) {
+    FD_LOG_ERR( ( "fd_alloc_join failed" ) ); 
+  }
+
+  fd_valloc_t valloc = fd_alloc_virtual( alloc );
+
+  /* Gossip set up */
+
+  // TODO: actually get a reasonable seed
+  ulong seed = 42;
+  ctx->repair = fd_repair_join( fd_repair_new( ctx->repair, seed, valloc ) );
+
+  FD_LOG_NOTICE(( "repair my addr - intake addr: %s, serve_addr: %s", tile->repair.repair_my_intake_addr, tile->repair.repair_my_serve_addr ));
+  FD_TEST( resolve_hostport( tile->repair.repair_my_intake_addr, &ctx->repair_my_intake_addr ) );
```

Comment:
I don't understand why this parses a string like "host:port" when the config definition has a separate ip address and port fields? It doesn't look like the format is what it expects?

Reply:
My repair and gossip implementations use host:port

Reply:
this is fixed on the integration branch for both gossip and repair. recommend leaving as is and the follow up will come very shortly

Reply:
I don't think pulling changes ad-hoc from a branch which is still a wip is a good idea. Suggest coming back to this in the follow up PR.